File: token-validation-core/src/main/java/no/nav/security/token/support/core/api/ProtectedWithClaims.java
package no.nav.security.token.support.core.api;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.METHOD;
import static java.lang.annotation.ElementType.TYPE;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Retention(RUNTIME)
@Target({ TYPE, METHOD })
@Protected
public @interface ProtectedWithClaims {

	String issuer();
	/**
	 * Required claims in token in key=value format.
     * If the value is an asterisk (*), it checks that the required key is present.
	 * @return array containing claims as key=value
	 */
	String[] claimMap() default {};

	/**
	 * How to check for the presence of claims,
	 * default is false which will require all claims in the list
	 * to be present in token. If set to true, any claim in the list
	 * will suffice.
	 *
	 * @return boolean
	 */
	boolean combineWithOr() default false;
}

File: token-validation-core/src/main/kotlin/no/nav/security/token/support/core/api/ProtectedWithClaims.kt
package no.nav.security.token.support.core.api

import no.nav.security.token.support.core.utils.Cluster
import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.annotation.AnnotationTarget.CLASS
import kotlin.annotation.AnnotationTarget.FUNCTION
import kotlin.annotation.AnnotationTarget.PROPERTY_GETTER
import kotlin.annotation.AnnotationTarget.PROPERTY_SETTER

@Retention(RUNTIME)
@Target(CLASS, FUNCTION, PROPERTY_GETTER, PROPERTY_SETTER)
@Protected
@MustBeDocumented
annotation class ProtectedWithClaims(val issuer : String,
                                     /**
                                      * Required claims in token in key=value format.
                                      * If the value is an asterisk (*), it checks that the required key is present.
                                      * @return array containing claims as key=value
                                      */
                                     val claimMap : Array<String> = [], val excludedClusters : Array<Cluster> = [],
                                     /**
                                      * How to check for the presence of claims,
                                      * default is false which will require all claims in the list
                                      * to be present in token. If set to true, any claim in the list
                                      * will suffice.
                                      *
                                      * @return boolean
                                      */
                                     val combineWithOr : Boolean = false)